### ISSUE TITLE
Building using a custom LIBVTERM_INCLUDE_DIR doesn't work

### ISSUE THREAD

**Issue by foleyfactor** [neovim/neovim#23237]
### Problem

When building Neovim 0.9.0 from source and using a custom LIBVTERM_INCLUDE_DIR, the include dir isn't properly being passed along. 

I believe this is because of the `target_link_libraries` invocation [here](https://github.com/neovim/neovim/blob/master/src/nvim/CMakeLists.txt#L35) has no matching `target_include_directories`. I'm able to build with a custom LIBLUV_INCLUDE_DIR because it calls `target_include_directories` [like this](https://github.com/neovim/neovim/blob/master/src/nvim/CMakeLists.txt#L25).

I think we should change the big `target_link_libraries` into multiple smaller invocations to `target_link_libraries` and `target_include_directories`, following the pattern of libluv. I'll create a PR that makes that change.

### Steps to reproduce

Make sure that libvterm's include directory isn't accessible in the default locations (there's probably a way to do this with environment variables, but I'm really unfamiliar with cmake). Then from a clean clone of the repo, do:

```
mkdir build 
pushd build

cmake3 \
  -DLIBVTERM_INCLUDE_DIR="path/to/libvterm/include/dir"
  ..

cmake3 --build .
```

When it attempts to build `auto/terminal.c.generated.h, ../../include/terminal.h.generated.h`, it fails:

```
[ 45%] Generating auto/terminal.c.generated.h, ../../include/terminal.h.generated.h
/tmp/tmp.zavDvK9OYG/neovim-0.9.0/src/nvim/terminal.c:46:19: fatal error: vterm.h: No such file or directory
 #include <vterm.h>
                   ^
compilation terminated.
gmake[2]: *** [src/nvim/auto/terminal.c.generated.h] Error 1
gmake[1]: *** [src/nvim/CMakeFiles/nvim.dir/all] Error 2
gmake: *** [all] Error 2
```

### Expected behavior

The `find_package` command is succeeding, so I would expect it to be able to find `vterm.h` and build successfully.

In fact, if I also include the path to `LIBVTERM_INCLUDE_DIR` in `LIBLUV_INCLUDE_DIR`, compilation succeeds.

### Neovim version (nvim -v)

0.9.0 commit 040f145

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

CentOS 7.9

### Terminal name/version

N/A

### $TERM environment variable

N/A

### Installation

Build from repo

**Comment by foleyfactor** [comment]
Whoops, looks like my suggestion doesn't quite work, after seeing the CI build failures in that PR. My cmake knowledge is super limited, so I don't know what the actual correct thing to do is, but I hope it at least gives you an idea for what I was hoping it would accomplish.

**Comment by dundargoc** [comment]
What are you trying to accomplish? Cmake should be able to find libvterm on its own if it's in a "regular" path, like when installing it with a distros package manager.

**Comment by foleyfactor** [comment]
Unfortunately, I'm unable to upgrade the system version of libvterm to 0.3.0 as it breaks some other things. I have a local copy of libvterm 0.3.0 that I'd like to point Neovim at for building instead.

**Comment by foleyfactor** [comment]
Sorry, I was away from my computer this weekend. That change looks to have fixed the issue, thanks for the quick response!

Could we backport this fix to 0.9.0? Happy to help out with that if possible.
